packages/node_modules/@node-red/runtime/lib/module/template/subflow.js.mustache

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function(RED) {
+    const fs = require('fs');
+    const path = require("path");
+    const crypt = require("crypto-js");
+
+    const file = path.join(__dirname, "subflow.json");
+    const text = fs.readFileSync(file);
+    const flow = JSON.parse(text);
+
+    function getDecoder(encoding) {
```

Comment:
Why is all the encoder/decoding logic here? Why is that not in the core of Node-RED?

If there are *any* bugs in here, we won't be able to fix them once users start publishing their own subflows containing this code. It also means we can never change how the encoding/decoding works under the covers as this code is no longer under our control.

For example, this would make it impossible to ever move to plugins.

Reply:
Moved the decoding process to the runtime side, in `registerSubflow`.